packages/cspell/src/util/reporters.ts

```diff
@@ -0,0 +1,50 @@
+import type { CSpellReporterModule, FileSettings, ReporterSettings } from 'cspell-lib';
+import { CSpellReporter } from '@cspell/cspell-types';
+
+function mergeEmitters<T extends keyof CSpellReporter>(
+    reporters: ReadonlyArray<CSpellReporter>,
+    emitterName: T
+): CSpellReporter[T] {
+    return (...args: unknown[]) => {
```

Comment:
You can determine the type of the arguments.
```
(...args: Parameters<CSpellReporter[T]>)
```

Reply:
I've tried that. ts throws a strange error
```
Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'CSpellReporter[T]'.
  Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'SpellingErrorEmitter & MessageEmitter & DebugEmitter & ErrorEmitter & ProgressEmitter & ResultEmitter'.
    Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'SpellingErrorEmitter'.
      Types of parameters 'args' and 'issue' are incompatible.
        Type '[issue: Issue]' is not assignable to type 'Parameters<CSpellReporter[T]>'.ts(2322)
```
which I could not fix so I left `unknown[]` here

Reply:
I've tried that. ts throws a strange error
```
Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'CSpellReporter[T]'.
  Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'SpellingErrorEmitter & MessageEmitter & DebugEmitter & ErrorEmitter & ProgressEmitter & ResultEmitter'.
    Type '(...args: Parameters<CSpellReporter[T]>) => Promise<void>' is not assignable to type 'SpellingErrorEmitter'.
      Types of parameters 'args' and 'issue' are incompatible.
        Type '[issue: Issue]' is not assignable to type 'Parameters<CSpellReporter[T]>'.ts(2322)
```
which I could not fix so I left `unknown[]` here